marlowe-playground-client/src/JavascriptEditor/State.purs

```diff
@@ -134,6 +141,37 @@ handleAction _ (InitJavascriptProject prunedContent) = do
   editorSetValue prunedContent
   liftEffect $ LocalStorage.setItem jsBufferLocalStorageKey prunedContent
 
+handleAction settings AnalyseContract = compileAndAnalyze settings (WarningAnalysis Loading) $ analyseContract settings
+
+handleAction settings AnalyseReachabilityContract =
+  compileAndAnalyze settings (ReachabilityAnalysis AnalysisNotStarted)
+    $ analyseReachability settings
+
+handleAction settings AnalyseContractForCloseRefund =
+  compileAndAnalyze settings (CloseAnalysis AnalysisNotStarted)
+    $ analyseClose settings
+
+-- This function runs a static analysis to the compiled code. It calls the compiler if
+-- it wasn't runned before and it switches to the error panel if the compilation failed
+compileAndAnalyze ::
+  forall m.
+  MonadAff m =>
+  SPSettings_ SPParams_ ->
+  AnalysisState ->
+  (Contract -> HalogenM State Action ChildSlots Void m Unit) ->
+  HalogenM State Action ChildSlots Void m Unit
+compileAndAnalyze settings initialAnalysisState doAnalyze = do
+  compilationResult <- use _compilationResult
+  case compilationResult of
+    NotCompiled -> do
+      -- The initial analysis state allow us to show an "Analysing..." indicator
+      assign _analysisState initialAnalysisState
+      handleAction settings Compile
+      compileAndAnalyze settings initialAnalysisState doAnalyze
+    (CompiledSuccessfully (InterpreterResult interpretedResult)) -> doAnalyze interpretedResult.result
+    (CompilationError _) -> handleAction settings $ BottomPanelAction $ BottomPanel.ChangePanel ErrorsView
```

Comment:
Also extra parenthesis here. There are some more in other files

Reply:
Changed all I saw 👍🏻